code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -133,36 +135,34 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 
 	var (
 		amount currency.Coin
-		n      currency.Coin
 		share  currency.Coin
 	)
 	share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
 	if err != nil {
 		err = errors.Wrap(err, fmt.Sprintf("%s, DistributeCoin operation, %s", code, info))
 		return
 	}
-	n, err = currency.Int64ToCoin(int64(len(payload.Signatures)))
+
+	amount, err = currency.MinusCoin(payload.Amount, share)
 	if err != nil {
-		err = errors.Wrap(err, fmt.Sprintf("%s, convert len signatures to coin, %s", code, info))
+		err = errors.Wrap(err, fmt.Sprintf("%s, payload.Amount - share * len(signatures), %s", code, info))
 		return
 	}
-	amount, err = currency.MinusCoin(payload.Amount, share*n)
+	payload.Amount = amount
+
+	rand.Seed(int64(trans.CreationDate.Duration()))
```

Comment:
User could manipulate the transaction creation date to always reward specific authorizer, so let's use the random round seed of the block if needed. We can get it from `ctx.GetBlock().GetRandomSeed()`.

Reply:
Fixed